botocore/client.py

```diff
@@ -164,6 +167,21 @@ def _determine_region_name(self, endpoint_config, region_name=None,
 
         return region_name
 
+    def _inject_s3_configuration(self, config_kwargs, scoped_config,
+                                 client_config):
+        s3_configuration = None
+
+        # Check the scoped config first
+        if scoped_config is not None:
+            s3_configuration = scoped_config.get('s3')
+
+        # Next the client config value takes precedence
+        if client_config is not None:
```

Comment:
That's an interesting point.  Should we maybe be merging the configs if a user has both scoped config and client config?  As we add more s3 options, it would mean that a customer would typically have to grab the s3 config from the session, merge their options on top, and then create a Config object.